borg/testsuite/repository.py

```diff
@@ -156,17 +156,6 @@ def test_crash_before_compact_segments(self):
             self.assert_equal(len(self.repository), 3)
             self.assert_equal(self.repository.check(), True)
 
-    def test_replay_of_readonly_repository(self):
-        self.add_keys()
-        for name in os.listdir(self.repository.path):
-            if name.startswith('index.'):
-                os.unlink(os.path.join(self.repository.path, name))
-        with patch.object(UpgradableLock, 'upgrade', side_effect=LockFailed) as upgrade:
-            self.reopen()
-            with self.repository:
-                self.assert_raises(LockFailed, lambda: len(self.repository))
-                upgrade.assert_called_once_with()
-
```

Comment:
I think it still does.
* Do a borg list with a remote repo that had it’s last writer crash and no other access since then
* client eventually calls Repository.get
* which calls Repository.get_transaction_id()
* which checks for crash by index_transaction_id != segments_transaction_id
* and calls Repository.replay_segments
* which calls self.prepare_txn
* which needs to upgrade

This is the one legitimate case i can come up where an always read only operation (that correctly uses an non exclusive lock) can get into prepare_txn.